core/src/main/java/nu/yona/server/analysis/entities/ActivityRepository.java

```diff
@@ -4,10 +4,22 @@
  *******************************************************************************/
 package nu.yona.server.analysis.entities;
 
+import java.time.LocalDateTime;
+import java.util.UUID;
+
+import org.springframework.data.jpa.repository.Query;
 import org.springframework.data.repository.CrudRepository;
+import org.springframework.data.repository.query.Param;
 import org.springframework.stereotype.Repository;
 
 @Repository
 public interface ActivityRepository extends CrudRepository<Activity, Long>
 {
+	@Query("select a from Activity a"
+			+ " where a.dayActivity = :dayActivity and a.activityCategory.id = :activityCategoryId and a.app = :app and"
+			+ " ((:startTime >= a.startTime and :startTime <= a.endTime) or" // New activity started during existing activity
+			+ " (:endTime >= a.startTime and :endTime <= a.endTime) or" // New activity ended during existing activity
+			+ " (a.startTime >= :startTime and a.endTime <= :endTime))") // Existing activity occurred during new activity
+	Activity findOverlapping(@Param("dayActivity") DayActivity dayActivity, @Param("activityCategoryId") UUID activityCategoryId,
```

Comment:
Currently this query returns 1 result, but there could be multiple overlapping existing activities. How do we handle that case? It would be worth a unit test...

Reply:
Only in theory, with the new code. Writing a unit test might not be easy.

Reply:
OK. I was thinking of the case when you have activity from App A from 12:05-12:10 and 12:12-12:15, and get new activity from App A from 12:00-12:20. But indeed there would be something seriously wrong with the Yona mobile app in that case :).

Reply:
And it won't be a disaster, as the deduplication now works properly.